File: src/utils/secure-storage.ts
/**
 * Wrapper para almacenamiento seguro de credenciales.
 * - Nativo: Keychain (iOS) / EncryptedSharedPreferences (Android)
 * - Web: localStorage (fallback para desarrollo)
 */
import { Platform } from 'react-native';

const TOKEN_KEY = 'evora_auth_token';
const USER_KEY = 'evora_user_data';

// En web usamos localStorage como fallback (solo para desarrollo)
const isWeb = Platform.OS === 'web';

async function getSecureStore() {
  if (isWeb) return null;
  return await import('expo-secure-store');
}

export async function saveToken(token: string): Promise<void> {
  if (isWeb) {
    localStorage.setItem(TOKEN_KEY, token);
    return;
  }
  const SecureStore = await getSecureStore();
  await SecureStore?.setItemAsync(TOKEN_KEY, token);
}

export async function getToken(): Promise<string | null> {
  if (isWeb) {
    return localStorage.getItem(TOKEN_KEY);
  }
  const SecureStore = await getSecureStore();
  return (await SecureStore?.getItemAsync(TOKEN_KEY)) ?? null;
}

export async function removeToken(): Promise<void> {
  if (isWeb) {
    localStorage.removeItem(TOKEN_KEY);
    return;
  }
  const SecureStore = await getSecureStore();
  await SecureStore?.deleteItemAsync(TOKEN_KEY);
}

export async function saveUserData(user: object): Promise<void> {
  if (isWeb) {
    localStorage.setItem(USER_KEY, JSON.stringify(user));
    return;
  }
  const SecureStore = await getSecureStore();
  await SecureStore?.setItemAsync(USER_KEY, JSON.stringify(user));
}

export async function getUserData<T>(): Promise<T | null> {
  if (isWeb) {
    const data = localStorage.getItem(USER_KEY);
    return data ? JSON.parse(data) : null;
  }
  const SecureStore = await getSecureStore();
  const data = await SecureStore?.getItemAsync(USER_KEY);
  return data ? JSON.parse(data) : null;
}

export async function clearAll(): Promise<void> {
  if (isWeb) {
    localStorage.removeItem(TOKEN_KEY);
    localStorage.removeItem(USER_KEY);
    return;
  }
  const SecureStore = await getSecureStore();
  await SecureStore?.deleteItemAsync(TOKEN_KEY);
  await SecureStore?.deleteItemAsync(USER_KEY);
}

File: lib/core/secure_storage.dart
import 'dart:convert';
import 'package:flutter/foundation.dart' show kIsWeb;
import 'package:flutter_secure_storage/flutter_secure_storage.dart';
import 'package:shared_preferences/shared_preferences.dart';

/// Wrapper para almacenamiento seguro de credenciales.
/// - Android/iOS: usa flutter_secure_storage (Keychain / EncryptedSharedPreferences)
/// - Web: usa shared_preferences (localStorage) como fallback
class SecureStorage {
  static const _storage = FlutterSecureStorage(
    aOptions: AndroidOptions(encryptedSharedPreferences: true),
  );

  static const _tokenKey = 'evora_auth_token';
  static const _userKey = 'evora_user_data';

  static Future<void> saveToken(String token) async {
    if (kIsWeb) {
      final prefs = await SharedPreferences.getInstance();
      await prefs.setString(_tokenKey, token);
    } else {
      await _storage.write(key: _tokenKey, value: token);
    }
  }

  static Future<String?> getToken() async {
    if (kIsWeb) {
      final prefs = await SharedPreferences.getInstance();
      return prefs.getString(_tokenKey);
    }
    return await _storage.read(key: _tokenKey);
  }

  static Future<void> removeToken() async {
    if (kIsWeb) {
      final prefs = await SharedPreferences.getInstance();
      await prefs.remove(_tokenKey);
    } else {
      await _storage.delete(key: _tokenKey);
    }
  }

  static Future<void> saveUserData(Map<String, dynamic> user) async {
    if (kIsWeb) {
      final prefs = await SharedPreferences.getInstance();
      await prefs.setString(_userKey, jsonEncode(user));
    } else {
      await _storage.write(key: _userKey, value: jsonEncode(user));
    }
  }

  static Future<Map<String, dynamic>?> getUserData() async {
    String? data;
    if (kIsWeb) {
      final prefs = await SharedPreferences.getInstance();
      data = prefs.getString(_userKey);
    } else {
      data = await _storage.read(key: _userKey);
    }
    if (data == null) return null;
    return jsonDecode(data) as Map<String, dynamic>;
  }

  static Future<void> clearAll() async {
    if (kIsWeb) {
      final prefs = await SharedPreferences.getInstance();
      await prefs.remove(_tokenKey);
      await prefs.remove(_userKey);
    } else {
      await _storage.delete(key: _tokenKey);
      await _storage.delete(key: _userKey);
    }
  }
}
